### ISSUE TITLE
Next-Cloudinary still cannot access NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME

### ISSUE THREAD

**Issue by reannu123** [expatfile/next-runtime-env#109]
I have properly set-up the <PublicEnvScript/> in the root layout, and I have properly set the environment variable in running a container, but I still get this error 
`Error: A Cloudinary Cloud name is required, please make sure NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME is set and configured in your environment.`

But in my .env file where I use with docker compose to run a container, I have already set it there

NEXT_PUBLIC_CLERK_SIGN_IN_URL=/sign-in
NEXT_PUBLIC_CLERK_SIGN_UP_URL=/sign-up
NEXT_PUBLIC_CLERK_AFTER_SIGN_IN_URL=/
NEXT_PUBLIC_CLERK_AFTER_SIGN_UP_URL=/
NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY=pk_test_bmVhdC1wYW5kYS0xMi5jbGVyay5hY2NvdW50cy5kZXYk
NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME="cloudnamehere"

The cloud name works in development

Also, when I build the NextJs application with a NEXT_CLOUDINARY_CLOUD_NAME variable, I do not get an error, but when I put a different value for that variable in running a container, it still uses the old variable, so in essence, it is not able to read the variable I put at run-time

**Comment by HofmannZ** [comment]
@reannu123 - Can you tell us a bit more about your build pipeline?

1. Are you using the app or pages directory?
2. What version of Next.js and `next-runtime-env` are you using?
3. Are you setting the env vars during the build, during runtime or both?
4. In which file are you configuring `next-runtime-env` and where are you configuring Cloudinary/Clerk?
4. How are you accessing the env vars? Using `process.env.` or using the `env()` util?

**Comment by reannu123** [comment]
Hello @HofmannZ thanks for responding,

1. Im using app directory
2. Next.js 14.1.0 and the latest version of `next-runtime-env` (3.2.1)
3. Im trying to set the vars at build time and runtime (docker run). When I set `NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME` at build time, I do not get an error but the runtime vars I set for it is not being read (it is using the var I set at build time, which I do not want). When I don't set any var at build time, but I set the var at runtime, i get a client exception error stating that I do not have the Cloud name set. So in conclusion, it's not reading the var I set at runtime.

The weird thing is, the other `NEXT_PUBLIC` variables work as intended. 
With `next-runtime-env`, it works when I change the clerk `NEXT_PUBLIC` vars (doesn't work without `next-runtime-env`), but the one used by next-cloudinary doesn't work for some reason.

4. Sorry I do not understand, I just followed the set-up where I place the `PublicEnvScript` in the head tag in the main layout.tsx,
5. However, the env vars are being accessed by the libraries that use them, so I'm not sure how Next-Cloudinary accesses the variables, but Clerk does it properly.
